rts/motoko-rts-tests/src/gc.rs

```diff
@@ -0,0 +1,274 @@
+// Naming conventions:
+//
+// - offset = index in the "heap" array/slice/vector
+// - address = address in the process's address space
+//
+// To convert an offset into an address, add heap array's address to the offset.
+
+mod heap;
+mod utils;
+
+use heap::MotokoHeap;
+use utils::{read_word, ObjectIdx, GC, GC_IMPLS, WORD_SIZE};
+
+use motoko_rts::gc::copying::copying_gc_internal;
+use motoko_rts::gc::mark_compact::compacting_gc_internal;
+use motoko_rts::types::*;
+
+use std::collections::{HashMap, HashSet};
+use std::fmt::Write;
+
+pub fn test() {
+    println!("Testing garbage collection ...");
+
+    // TODO: Add more tests
```

Comment:
A nice one to tru might be some large binary tree with a linear number of nodes due to sharing, eg:
```
let p0 = ();
let p1= (p0, p0);
...
let pn = (pn-1, pn-1);
```
and then maybe a cyclic version of that.

Reply:
Noted, I will add this case when I open a PR to add tests.

Reply:
> Indeed, I wonder if adding some monadic syntax to let you build heap graphs using monadic syntax would be useful for writing tests. Does Rust let you do that easily?

Not easily possible with Rust syntax, but we could use macros.